File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for i in range(len(s)):
        if s[i] == '(':
            stack.append(s[i])
        elif len(stack) != 0 and s[i] == ')' and stack[-1] == '(':
            stack.pop()
        elif len(stack) == 0 and s[i] == ')':
            stack.append(s[i])
    if len(stack) == 0:
        answer = True
    else:
        answer = False
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    const bucket = [];
    
    for (let i = 0; i<s.length; i++){
        if(s[i] == "("){
            bucket.push(s[i])
        }
        if(s[i] == ")"){
            if(bucket.length === 0) return false;
            bucket.pop()
        }
    }
    
    if(bucket.length == 0) return true
    else return false
    
    return answer;
}